py/server/deephaven/barrage.py

```diff
@@ -68,21 +75,49 @@ def barrage_session(host: str,
         else:
             target_uri = f"dh+plain://{target_uri}"
 
-        _j_client_config_builder = _JClientConfig.builder()
-        _j_client_config_builder.target(_JDeephavenTarget.of(_JURI(target_uri)))
-        if tls_root_certs:
-            _j_ssl_config =_JSSLConfig.builder().trust(_JTrustCustom.ofX509(tls_root_certs, 0, len(tls_root_certs))).build()
-            _j_client_config_builder.ssl(_j_ssl_config)
-        _j_client_config = _j_client_config_builder.build()
+        j_client_config = _build_client_config(target_uri, tls_root_certs)
         auth = f"{auth_type} {auth_token}"
 
-        _j_barrage_session_factory = j_barrage_session_factory_client.factory(_j_client_config, auth)
-        return BarrageSession(_j_barrage_session_factory.newBarrageSession())
+        try:
+            return _get_barrage_session_uri(j_client_config, auth)
+        except:
+            return _get_barrage_session_direct(j_client_config, auth)
     except Exception as e:
-        raise DHError(e,"failed to get a barrage session to the target remote Deephaven server.") from e
+        raise DHError(e, "failed to get a barrage session to the target remote Deephaven server.") from e
+
+
+def _get_barrage_session_uri(client_config, auth) -> BarrageSession:
+    j_barrage_session_factory_client = uri.resolve(
+        "dh:///app/io.deephaven.server.barrage.BarrageSessionFactoryClient/field/instance")
+    j_barrage_session_factory = j_barrage_session_factory_client.factory(client_config, auth)
+    return BarrageSession(j_barrage_session_factory.newBarrageSession())
+
+
+def _get_barrage_session_direct(client_config, auth) -> BarrageSession:
```

Comment:
It's worth noting that while this direct construct works, it does _not_ share any of the state or configuration that the server provides; namely, when you are doing it with the server context it provides a singleton executor, allocator, and outbound SSL configuration.  Additionally, it allows the server to hook in additional channel building options.